lib/secure_headers/railtie.rb

```diff
@@ -3,9 +3,17 @@
   module SecureHeaders
     class Railtie < Rails::Engine
       isolate_namespace ::SecureHeaders if defined? isolate_namespace # rails 3.0
+      conflicting_headers = ['X-Frame-Options', 'X-XSS-Protection', 'X-Content-Type-Options']
       initializer "secure_headers.action_controller" do
         ActiveSupport.on_load(:action_controller) do
           include ::SecureHeaders
+
+          unless Rails.application.config.action_dispatch.default_headers.nil?
```

Comment:
I feel like this would raise an error on rails < 4, default_headers wouldn't even be defined! Must be more rails method missing crap magic.